File: LogiOS/TrustarApp/Detect/SensorRecord.swift
//
//  SensorRecord.swift
//  TrustarApp
//
//  Created by DionSoftware on 2023/03/23.
//  Copyright © 2023 DionSoftware. All rights reserved.
//

import Foundation
class SensorRecord {
    class DropRecent : SensorRecord {
        var allocationNo: String
        var from: Int64
        var to: Int64
        var isFinished: Bool
        init(allocationNo: String, from: Int64, to: Int64, isFinished: Bool) {
            self.allocationNo = allocationNo
            self.from = from
            self.to = to
            self.isFinished = isFinished
        }
    }
    
    class Record : SensorRecord {
        var event: SensorDetectEvent
        init(event: SensorDetectEvent) {
            self.event = event
        }
    }
}

File: LogAndroid/TrustarApp/src/main/kotlin/jp/co/toukei/log/trustar/detect/SensorRecord.kt
package jp.co.toukei.log.trustar.detect

import jp.co.toukei.log.trustar.db.user.entity.SensorDetectEvent

sealed class SensorRecord {

    class DropRecent(
        @JvmField val allocationNo: String,
        @JvmField val from: Long,
        @JvmField val to: Long,
        @JvmField val isFinished: Boolean,
    ) : SensorRecord()

    class Record(
        @JvmField val event: SensorDetectEvent,
    ) : SensorRecord()
}
